.github/workflows/build.yml

```diff
@@ -47,3 +48,126 @@ jobs:
 
     - name: Test run
       run: /usr/local/bin/rtl_airband -v
+    
+    - name: Set snapshot version
+      id: sha
+      run: echo "snapshot_version=$(script/snapshot_version.sh)" >> $GITHUB_OUTPUT
+    
+    - name: Create debian package
+      if: runner.os == 'Linux'
+      run: |
+        pkg_dir="$(mktemp --directory)"
+        export PACKAGE="rtl-airband"
+        export VERSION="${{ steps.sha.outputs.snapshot_version }}"
+        export ARCH="$(dpkg --print-architecture)"
+        export MAINTAINER="charlie-foxtrot"
+        export DEPENDS="libc6 (>= 2.34), libconfig++9v5 (>= 1.5-0.4), libmp3lame0 (>= 3.100), libshout3 (>= 2.4.5), librtlsdr0 (>= 0.6.0), libsoapysdr0.8 (>= 0.8.1), libfftw3-single3 (>= 3.3.8), libpulse0 (>= 14.2)"
+        export DESCRIPTION="RTLSDR Airband - A multiband decoder for AM and NFM signals"
+        script/package.sh "${pkg_dir}"
+    
+    # TODO use brew for macOS? https://brew.sh/
+    - name: Create macOS package
+      if: runner.os == 'macOS'
+      run: |
+        tar -zcvf rtl-airband_${{ steps.sha.outputs.snapshot_version }}_${{ matrix.os }}.tar.gz build_Release_NFM/src/rtl_airband
+    
+    - name: "Upload package as artifact"
+      uses: actions/upload-artifact@v3
+      with:
+        name: "rtl-airband_${{ steps.sha.outputs.snapshot_version }}_${{ matrix.os }}"
+        path: |
+          *.deb
+          *.tar.gz
+        retention-days: 7
+
+  build-foreign-arch:
```

Comment:
what is this doing?  its running on a linux host and using docker to build `amd64`, `arm64`, `arm/v7`, and `arm/v6`?  how does it differ from a matrix build? if it is independent can it be its own `.yml`?

Reply:
It can be put in its own yaml. As first version I put together because was easier to link this job as a dependency of the previous one, I'll split in a separate one.

I used this way to get the build for foreign architecture in an easy and reliable manner. Another one would be to use `gcc` cross-compile capability (and it may be also faster), but with cross-compilation there is a small chance that linking with dynamic library could have some issues.
Same thing could be done on macOS (amd64 and arm64) if needed.

The main advantages of building  by using this method are:
* the build environment is under our control and it can be replicated in the local machine, so no dependency on which configuration/package version is used by the workers or side effects, so it makes debugging the build much easier (as a matter of fact, the version of the libraries we are compiling against is based on the one available in that OS version, not as a requirement of the project).
* we are not limited to the architecture available on GitHub (e.g. RISC-V 32/64 bit in a future maybe?)
* it doesn't need to modify the worker, so better isolation and pre-configured images could also be used.

Reply:
I had a look at dependencies between workflows and it is possible to link them, **but** it will run only on default branch [see notes](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run).

What it can be done is to create a parent workflow that calls in order the child workflows. I'm not sure if it is really helpful for this size, thought.